cfme/tests/configure/test_tag.py

```diff
@@ -124,6 +126,29 @@ def test_map_tagging_crud(appliance, category, soft_assert):
                                           .format(map_tag_entity.label))
 
 
+@pytest.fixture
+def csv_tag_file(create_vm, category, tag):
+    temp_file = tempfile.NamedTemporaryFile(suffix=".csv")
+    csv_data = f'name,category,entry\n{create_vm.name},{category.display_name},{tag.display_name}'
+    # create CSV file
+    with open(temp_file.name, "w") as file:
+        file.write(csv_data)
+    yield file
+
+
+@test_requirements.tag
+def test_import_tag(appliance, create_vm, category, tag, csv_tag_file):
+    """Test importing tag via file
+    Polarion:
+        assignee: prichard
+        initialEstimate: 1/4h
+        casecomponent: Tagging
+    Bugzilla:
+        1792185
+    """
+    category.import_tag_from_file(csv_tag_file.name)
```

Comment:
I think some additional assertions are necessary here - you have the tag and category display names, you should be able to instantiate a Tag and assert that it exists.

Look at something like:
```
assert category.tags.instantiate(display_name=tag.display_name).exists
```

Reply:
If the tag does not exist the import will not work. The BZ issue was that an error gets displayed after "Upload" is clicked but before the apply. We are covering the BZ here. I guess I could add an assert that the tag exists on the VM?

Reply:
gotcha, that makes more sense - please update the docblock with more detailed description of what's being tested.

Also correct about the additional assertion of the tag on the VM.  You should be able to call `tag in create_vm.get_tags`, as the tag has an `__eq__` implementation.

Reply:
I put the assert against the vm tags in the category.import_tag_from_file() method.
I could have put it in the test case itself, but chose the method.
If I put it in the testcase I wouldn't need to pass down the extra 2 args to import_tag_from_file and could leave more flexibility in other tests to check results for an already existing tag (just comparing the display name strings) without having to create the tag object.
But how much more tagging tests are we going to write? And if we do write more, couldn't I just move the assert out then?

Reply:
I moved the validation of the tag on the VM out into the testcase.